jupyter_kernel_mgmt/subproc/launcher.py

```diff
@@ -44,13 +45,14 @@ class SubprocessKernelLauncher:
     """
     transport = 'tcp'
 
-    def __init__(self, kernel_cmd, cwd, extra_env=None, ip=None):
+    def __init__(self, kernel_cmd, cwd, extra_env=None, ip=None, launch_params=None):
```

Comment:
I'd probably say this level of the API should have specific kwargs which the provider is responsible for extracting from the generic `launch_params`. E.g. this could be called `argv_format_args` or something.

Reply:
Good point - I'll make this change.

Reply:
When I first read this, I figured you were stating that these three keyword parameters could be collapsed into **kwargs.  But now that I look closer, it seems like your comment really applies to the code that constructs the `SubprocessKernelLauncher` instance, since _that_ would be location in which argv-only parameters could be extracted into `argv_format_args`.  Or are you saying that this method would have additional code that pulls the argv-only parameters into `argv_format_args`, for example?

I don't mind replacing this single parameter with **kwargs, but I'd still need to add code that pulls `launch_params` out of `kwargs` making those values available to the format code until the metadata definition is formalized.  Or update the callers to use a different keyword. 

Could you please clarify what you mean?  Sorry.